Installation issue: llvm fails to build assembler files with CMake 3.19.0

### Steps to reproduce the issue

```console
$ spack install llvm
...
-- Performing Test LIBCXX_SUPPORTS_WEXTRA_SEMI_FLAG
/spack-stage-dir/202269/root/spack-stage-llvm-11.0.0-stpte4xqkfbkbahzzcdnqw5pvxrjm6hs/spack-src/compiler-rt/lib/tsan/rtl/tsan_rtl_amd64.S:7:1: error: expected identifier or '(' before '.' token
 .section .text
 ^
/spack-stage-dir/202269/root/spack-stage-llvm-11.0.0-stpte4xqkfbkbahzzcdnqw5pvxrjm6hs/spack-src/compiler-rt/lib/tsan/rtl/tsan_rtl_amd64.S:16:5: error: invalid preprocessing directive #Save
   # Save scratch registers.
     ^~~~
/spack-stage-dir/202269/root/spack-stage-llvm-11.0.0-stpte4xqkfbkbahzzcdnqw5pvxrjm6hs/spack-src/compiler-rt/lib/tsan/rtl/tsan_rtl_amd64.S:44:5: error: invalid preprocessing directive #Align
   # Align stack frame.
     ^~~~~
...
make: *** [Makefile:174: all] Error 2
==> [2020-11-25-20:09:46.809565] Error: ProcessError: Command exited with status 2:
```

This is a combination of both an LLVM and CMake issue:

LLVM bug: https://bugs.llvm.org/show_bug.cgi?id=48234
LLVM fix: https://reviews.llvm.org/D86308
CMake bug: https://gitlab.kitware.com/cmake/cmake/-/issues/21469
CMake fix: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5519

This is a placeholder issue that will be closed by the above referenced patches being applied

--- a/var/spack/repos/builtin/packages/llvm/package.py
+++ b/var/spack/repos/builtin/packages/llvm/package.py
@@ -200,6 +200,11 @@ class Llvm(CMakePackage, CudaPackage):
             "system debug server",
     )
 
+    # LLVM bug https://bugs.llvm.org/show_bug.cgi?id=48234
+    # CMake bug: https://gitlab.kitware.com/cmake/cmake/-/issues/21469
+    # Fixed in upstream versions of both
+    conflicts('^cmake@3.19.0', when='@6.0.0:11.0.0')
+
     # Github issue #4986
     patch("llvm_gcc7.patch", when="@4.0.0:4.0.1+lldb %gcc@7.0:")
     # Backport from llvm master + additional fix

